Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1223 @@
+        }
+        return undefined;
+    }
+
+    function queryBooleanValue(node, tagName, namespace) {
+        var result = queryFirstNode(node, tagName, namespace);
+        return defined(result) ? result.textContent === '1' : undefined;
+    }
+
+    function resolveHref(href, proxy, sourceUri, uriResolver) {
+        if (!defined(href)) {
+            return undefined;
+        }
+        var hrefResolved = false;
+        if (defined(uriResolver)) {
+            var blob = uriResolver[href];
+            if (defined(blob)) {
+                hrefResolved = true;
+                href = blob;
+            }
+        }
+        if (!hrefResolved && defined(sourceUri)) {
+            var baseUri = new Uri(document.location.href);
+            sourceUri = new Uri(sourceUri);
+            href = new Uri(href).resolve(sourceUri.resolve(baseUri)).toString();
+            href = proxyUrl(href, proxy);
+        }
+        return href;
+    }
+
+    function processPositionGraphics(dataSource, entity) {
+        var label = entity.label;
+        if (!defined(label)) {
+            label = createDefaultLabel();
+            entity.label = label;
+        }
+        label.text = entity.name;
+
+        var billboard = entity.billboard;
+        if (!defined(billboard)) {
+            billboard = createDefaultBillboard();
+            entity.billboard = billboard;
+        }
+
+        if (!defined(billboard.image)) {
+            billboard.image = dataSource._pinBuilder.fromColor(Color.YELLOW, 64);
+        }
+
+        if (defined(billboard.scale)) {
+            var scale = billboard.scale.getValue();
+            if (scale !== 0) {
+                label.pixelOffset = new Cartesian2((scale * 16) + 1, 0);
+            } else {
+                //Minor tweaks to better match Google Earth.
+                label.pixelOffset = undefined;
+                label.horizontalOrigin = undefined;
+            }
+        }
+    }
+
+    function processPathGraphics(dataSource, entity) {
+        var path = entity.path;
+        if (!defined(path)) {
+            path = new PathGraphics();
+            path.leadTime = 0;
+            entity.path = path;
+        }
+
+        //
+        //        var polyline = styleEntity.polyline;
+        //        if (defined(polyline)) {
+        //            path.material = polyline.material;
+        //            path.width = polyline.width;
+        //        }
+    }
+
+    function createDefaultBillboard(proxy, sourceUri, uriResolver) {
+        var billboard = new BillboardGraphics();
+        billboard.width = BILLBOARD_SIZE;
+        billboard.height = BILLBOARD_SIZE;
+        billboard.scaleByDistance = new NearFarScalar(2414016, 1.0, 1.6093e+7, 0.1);
+        var DEFAULT_ICON = '../../../Build/Cesium/Assets/Textures/maki/marker.png';
+        billboard.image = resolveHref(DEFAULT_ICON, proxy, sourceUri, uriResolver);
+        return billboard;
+    }
+
+    function createDefaultPolygon() {
```

Comment:
I'll stop commenting on unused code, but in general if it's not being called (like this function), delete it.